src/sys/signal.rs

```diff
@@ -218,11 +256,88 @@ pub enum SigmaskHow {
     SIG_SETMASK = libc::SIG_SETMASK,
 }
 
+/// SigSet represents a set of signals and is a thin wrapper around the libc APIs for sigset_t.
+/// However, due to the clunky nature of the libc APIs around sigset_t (SigSet::add, SigSet::extend
+/// and so on), a more idiomatic way of constructing SigSets is possible using BitOr.
+///
+/// # Examples
+///
+/// ```
+/// # use ::sys::signal::*;
+/// let sigset_a = Signal::SIGTERM | Signal::SIGHUP;
+/// let sigset_b = sigset_a | Signal::SIGUSR1;
+/// let mut sigset_c = SigSet::empty();
+/// sigset_c |= Signal::SIGUSR2;
+/// sigset_c |= sigset_b;
+/// let sigset_d = sigset_c | sigset_a;
+/// ```
+///
+/// The above would traditionally require far more finicking with SigSet::add, SigSet::extend and
+/// so on. For further idiomatic ways to interact with SigSets see the sigset! macro.
 #[derive(Clone, Copy)]
 pub struct SigSet {
     sigset: libc::sigset_t
 }
 
+/// sigset is a macro that allows for far more idiomatic initalisation of SigSet. Note that while
```

Comment:
Yeah, `From<Signal> for SigSet` is a very reasonable thing to do IMO here.  Agreed that it would be better to not have that macro unless it is really required.